From d7b2d547b61b7d1cf7b4bbab42ad400719f96ae4 Mon Sep 17 00:00:00 2001
From: Xianzhong <xianzhong.li@nxp.com>
Date: Mon, 12 Nov 2018 21:07:47 +0800
Subject: [PATCH 5092/5242] MGS-4376: gpu: dts: increase reserved size to 256M
 for imx8

commit  6d8398135f101bcab8b22184c616086acdea2903 from
https://source.codeaurora.org/external/imx/linux-imx.git

alloc_contig_range easily return -EBUSY when try to isolate pages,
there are lots of messages with PFNs busy when run GPU tests.
[  622.370671] alloc_contig_range: [4ea70, 4ea7c) PFNs busy
[  626.518072] alloc_contig_range: [4ea90, 4ea9c) PFNs busy

these problems are related wht CMA migration for fragments,
need enlarge GPU reserved size to reduce CMA fragments.

Signed-off-by: Xianzhong <xianzhong.li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi      |    2 +-
 arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi      |    2 +-
 .../boot/dts/freescale/fsl-imx8qm-device.dtsi      |    2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi
index 4f430db..90a017c 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi
@@ -2650,7 +2650,7 @@
 	imx8_gpu_ss: imx8_gpu_ss {
 		compatible = "fsl,imx8qxp-gpu", "fsl,imx8-gpu-ss";
 		cores = <&gpu_3d0>;
-		reg = <0x0 0x80000000 0x0 0x80000000>, <0x0 0x0 0x0 0x8000000>;
+		reg = <0x0 0x80000000 0x0 0x80000000>, <0x0 0x0 0x0 0x10000000>;
 		reg-names = "phys_baseaddr", "contiguous_mem";
 		depth-compression = <0>;
 		status = "disabled";
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
index eec6bbb..7fa00f6 100755
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
@@ -1173,7 +1173,7 @@
 
 	gpu: gpu@38000000 {
 		compatible = "fsl,imx8mq-gpu", "fsl,imx6q-gpu";
-		reg = <0x0 0x38000000 0 0x40000>, <0x0 0x40000000 0x0 0xC0000000>, <0x0 0x0 0x0 0x8000000>;
+		reg = <0x0 0x38000000 0 0x40000>, <0x0 0x40000000 0x0 0xC0000000>, <0x0 0x0 0x0 0x10000000>;
 		reg-names = "iobase_3d", "phys_baseaddr", "contiguous_mem";
 		interrupts = <GIC_SPI 3 IRQ_TYPE_LEVEL_HIGH>;
 		interrupt-names = "irq_3d";
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi
index 6ef61a4..6913949 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi
@@ -3478,7 +3478,7 @@
 	imx8_gpu_ss: imx8_gpu_ss {
 		compatible = "fsl,imx8qm-gpu", "fsl,imx8-gpu-ss";
 		cores = <&gpu_3d0>, <&gpu_3d1>;
-		reg = <0x0 0x80000000 0x0 0x80000000>, <0x0 0x0 0x0 0x8000000>;
+		reg = <0x0 0x80000000 0x0 0x80000000>, <0x0 0x0 0x0 0x10000000>;
 		reg-names = "phys_baseaddr", "contiguous_mem";
 		status = "disabled";
 	};
-- 
1.7.9.5

